src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,122 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import nextstep.subway.exception.EmptySectionException;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST,
+            CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public void addSection(Section section) {
+        if (sections.isEmpty()) {
+            sections.add(section);
+            return;
+        }
+
+        if (insertSectionIsUpStation(section)) {
+            return;
+        }
+
+        if (insertSectionIsDownStation(section)) {
+            return;
+        }
+
+        if (insertSectionBetweenSections(section)) {
+            return;
+        }
+
+        throw new IllegalArgumentException("section 을 넣을 수 없습니다.");
+    }
+
+    private boolean insertSectionIsUpStation(Section section) {
+        if (sections.get(0).getUpStation().equals(section.getDownStation())) {
+            sections.add(0, section);
+            return true;
+        }
+        return false;
+    }
+
+    private boolean insertSectionIsDownStation(Section section) {
+        if (sections.get(sections.size() - 1).getDownStation().equals(section.getUpStation())) {
+            sections.add(sections.size() - 1, section);
+            return true;
+        }
+        return false;
+    }
+
+    private boolean insertSectionBetweenSections(Section registerSection) {
+        for (Section section : sections) {
+            int index = sections.indexOf(section);
+
+            if (section.getUpStation().equals(registerSection.getUpStation())) {
+                validateDistance(section, registerSection);
+
+                sections.remove(section);
+
+                sections.add(index, registerSection);
+                sections.add(index + 1
+                        , new Section(section.getLine(), section.getDownStation(),
+                                registerSection.getDownStation(),
```

Comment:
```suggestion
                sections.add(index, registerSection);
                sections.add(index + 1
                        , new Section(section.getLine(), registerSection.getDownStation(),
                                sections.get(index + 1).getDownStation(),
```

잘 이해가 안가는데, 바로 앞에 registerSection 이 있으면,
registerSection 의 downStation 이 다음 Section 의 upStation 이 되어야 하지 않나요?